### ISSUE TITLE
Add disabling workspace wrap

### ISSUE THREAD

**Issue by tsujp** [i3/i3#3900]
<pre>
[ ] Bug
[x] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior
Currently when I `workspace next` to the end workspace, and then I `workspace next` again it wraps back to the first workspace.

## Desired Behavior
. I do not want workspaces wrapping back to the start since it's extremely disorientating if done at either edge - focus-wrapping was added for moving focus between windows within a workspace, I am asking for similar but for the workspaces themselves.

## Environment
<pre>
Binary i3 version:  4.17.1 (2019-08-30) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.17.1 (2019-08-30) (pid 3529)abort…)
</pre>

<pre>
- Linux Distribution & Version: Arch Linux, 5.4.10-arch1-1
- Are you using a compositor (e.g., xcompmgr or compton): no
</pre>

**Comment by aste4** [comment]
+1 to this feature request. It is helpful when the first workspace is the most used one. Navigating to it can be done with "workspace prev" multiple times.

It also models the behavior of multiple spaces on MacOS

**Comment by tsujp** [comment]
> +1 to this feature request. It is helpful when the first workspace is the most used one. Navigating to it can be done with "workspace prev" multiple times.
> 
> It also models the behavior of multiple spaces on MacOS

Yeah my experience with Yabai first (which is amazing on macOS) and the behaviour of Spaces there definitely influenced my desire for this option here. Just being able to spam `next` `next` `next` in series and reach the end is better in a pinch.

**Comment by aspiers** [comment]
I agree. Alongside the ability to spam moving in one direction, another reason for wanting this is that if I'm on workspace 1 and I accidentally try `prev` thinking that I was actually on workspace 2, I don't want it to go to the end - I just want it to ignore my mistake.

**Comment by aspiers** [comment]
Seems like it would be a simple matter of adding an option and wrapping these lines in the corresponding conditional:

https://github.com/i3/i3/blob/e2b2a286251481abd83e72127a6f486753756c7e/src/workspace.c#L617-L618

https://github.com/i3/i3/blob/e2b2a286251481abd83e72127a6f486753756c7e/src/workspace.c#L683-L684

I guess something similar would be needed for `workspace_next_on_output` and `workspace_prev_on_output`.

**Comment by orestisfl** [comment]
I don't see a reason why not, just introduce a `--no-wrap` flag for the relevant commands.

**Comment by yatinlala** [comment]
Here is a simple [script](https://raw.githubusercontent.com/yatinlala/dotfiles/main/scripts/code/scripts/change-workspace) that you can use to get this functionality. Just replace `workspace prev` with `change-workspace prev` and `workspace next` with `change-workspace next` in your i3 config and add `change-workspace` to your path. You need [wmctrl](http://tripie.sweb.cz/utils/wmctrl/) for this to work.

**Comment by rfb3** [comment]
The link to the change-workspace script is dead. @yatinlala , is there another way for me to get a copy of your simple script and/or just a quick description of how it works so that I could try to craft my own? Thanks!

**Comment by yatinlala** [comment]
Sorry about the dead link @rfb3, here is a [working one](https://github.com/yatinlala/dotfiles/blob/2c286889dce89db3f3ca976692c8f93ea0308f5f/scripts/code/scripts/change-workspace). Let me know if you're having any other issues with it.

**Comment by rfb3** [comment]
No problem. Thanks for sharing. I've been wanting this behavior since I first tried i3.

It _mostly_ worked for me, but would still wrap around if I didn't have a workspace 10. I saw what you were doing and tweaked it slightly. Now works when my top window is, e. g., 9:

```
#! /bin/sh

case $1 in
    prev) wmctrl -d | head -1 | grep '\*' || i3-msg workspace prev;;
    next) wmctrl -d | tail -1 | grep '\*' || i3-msg workspace next;;
esac
```
